docs/graph_engine/data_spec.rst

```diff
@@ -17,9 +17,77 @@ Users can generate a graph in either format then our pipeline will convert it in
 DeepGNN also supports writing custom decoders, see [the decoders file](https://github.com/microsoft/DeepGNN/blob/main/src/python/deepgnn/graph_engine/snark/decoders.py).
 Just inheret the base class Decoder, overwrite the decode function and pass the new decoder as an argument to the converter or dispatcher.
 
-1. `JSON <#json-format>`_: Heterogeneous or homegeneous graph.
+1. `Linear <#linear-format>`_: Heterogeneous or homegeneous graph.
 
-2. `TSV <#tsv-format>`_: Homogeneous graph only.
+2. `JSON <#json-format>`_: Heterogeneous or homegeneous graph.
+
+3. `TSV <#tsv-format>`_: Homogeneous graph only.
+
+Linear Format
+===========
+
+Linear format is most directly what is given to binary writers, it supports nodes and their outgoing edges in
+different partitions and is smaller and faster to convert.
+
+`graph.linear` layout,
+
+.. code-block:: text
+	<node info>
+	<edge_1_info>
+	<edge_2_info>
+	...
+
+node_info: node_id -1 node_type node_weight <features>
+edge_info: src dst edge_type edge_weight <features>
+features[dense]: dtype_name length v1 v2 ... dtype_name2 length2 v1 v2 ...
+features[sparse]: dtype_name values.size,coords.shape[1] c1 c2 ... v1 v2 ...
+features[sparse]: dtype_name values.size,0 c1 c2 ... v1 v2 ...
```

Comment:
why there are 2 representations of sparse features? Could you add explanation to the doc?